src/SFA.DAS.ApprenticePortal/Application/Homepage/Queries/GetApprenticeHomepageQueryHandler.cs

```diff
@@ -0,0 +1,41 @@
+﻿using MediatR;
+using SFA.DAS.ApprenticePortal.Application.Extensions;
+using SFA.DAS.ApprenticePortal.InnerApi.ApprenticeAccounts.Requests;
+using SFA.DAS.ApprenticePortal.InnerApi.ApprenticeCommitments.Requests;
+using SFA.DAS.ApprenticePortal.Models;
+using SFA.DAS.SharedOuterApi.Configuration;
+using SFA.DAS.SharedOuterApi.Interfaces;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.ApprenticePortal.Application.Homepage.Queries
+{
+    public class GetApprenticeHomepageQueryHandler : IRequestHandler<GetApprenticeHomepageQuery, GetApprenticeHomepageQueryResult>
+    {
+        private readonly IApprenticeAccountsApiClient<ApprenticeAccountsApiConfiguration> _accountsApiClient;
+        private readonly IApprenticeCommitmentsApiClient<ApprenticeCommitmentsApiConfiguration> _commitmentsApiClient;
+
+        public GetApprenticeHomepageQueryHandler(
+            IApprenticeAccountsApiClient<ApprenticeAccountsApiConfiguration> accountsApiClient,
+            IApprenticeCommitmentsApiClient<ApprenticeCommitmentsApiConfiguration> commitmentsApiClient)
+            => (_accountsApiClient, _commitmentsApiClient) = (accountsApiClient, commitmentsApiClient);
+
+        public async Task<GetApprenticeHomepageQueryResult> Handle(GetApprenticeHomepageQuery request, CancellationToken cancellationToken)
+        {
+            var (_apprentice, _apprenticeships) = await TaskEx.AwaitAll(
+                Task.FromResult(_accountsApiClient.Get<Apprentice>(new GetApprenticeRequest(request.ApprenticeId))).Result,
+                Task.FromResult(_commitmentsApiClient.Get<GetApprenticeApprenticeshipsResult>(new GetApprenticeApprenticeshipsRequest(request.ApprenticeId))).Result
+                );
+
+            return new GetApprenticeHomepageQueryResult
+            {
+                apprenticeHomepage = new ApprenticeHomepage
+                {                    
+                    apprentice = _apprentice,
+                    apprenticeship = _apprenticeships?.apprenticeships.FirstOrDefault()
+                },                
+            };
```

Comment:
I might be wrong here, but I think if the apprentice personal details do not exist in the ApprenticeAccounts API, It will return a 404 response and this will result in a null object. Therefore we will lose the ApprenticeId if we model the ApprenticeHomePage like this.

Reply:
the apprentice id is the parameter sent in the querystring, the caller will know.  the api will not 404 but return a null object for the apprentice and apprenticeship in this instance

Reply:
Ive updated the controller - in the event that an accounts apprentice is null then it will now return a 404 NotFound response